### DISCUSSION TITLE
Support for chunked file uploads

### DISCUSSION THREAD

**Discussion by SamuelNitsche** [filamentphp/filament#47] [Ideas]
I saw that there is already a FileUpload field but this is only helpful for small to medium sized files. I am running a video platform where I upload files sized 400mb or bigger.

**Comment by jcsoriano** [comment]
+1, wondering if anyone has pointers on how to make it work with Filament?

**Comment by falconsmilie** [comment]
Came across this today. Cloudflare has a 100MB/200MB limit. Supporting chunking would be a nice feature to have.

**Comment by joshembling** [comment]
Is there any update on this?

**Comment by awaisali3405** [comment]
I am using filament file upload. I need file to converting into chunk on the uploading. Do anyone have solution to use chunk in file pond using filament?

**Comment by itsemon245** [comment]
I think chunk file upload is a must have option in order to upload large files

**Comment by votkapower** [comment]
Is there a package that can handle it?

**Comment by ifteakhor** [comment]
any updates on this?

**Comment by troyeisert** [comment]
+1 this feature would be very nice to have.

**Comment by unoriginalscreenname** [comment]
The included file upload component misses two really important features here.  1) it automatically starts uploading the file, you can't just drag and drop to select a file and there's no way to disable it.  2) any large file takes an very long time to upload as it doesn't appear to support chunked uploading.  I immediately ran into both of this problems right out of the gate and it's a huge disappointment.  I almost can't believe the file upload component doesn't support uploading files bigger than a small image.

**Reply by ifteakhor** [reply]
> 2) any large file takes an very long time to upload as it doesn't appear to support chunked uploading.  I immediately ran into both of this problems right out of the gate and it's a huge disappointment.  I almost can't believe the file upload component doesn't support uploading files bigger than a small image.  

You can upload large files by changing livewire config files, max upload file is set to 2MB or 12MB. I've changed it to 512MB but chunk uploading is better solution than this because my files are getting bigger and bigger

**Comment by zakmcrbit** [comment]
+1

**Comment by bobicloudvision** [comment]
+1

**Comment by peter-mw** [comment]
+1

**Comment by stuartcusackie** [comment]
+1

**Comment by GeminiDev1** [comment]
There's a package that supports chunk uploads that has been released recently:

https://filamentphp.com/plugins/ralphjsmit-upload-pro

**Reply by aliaswpeu** [reply]
That's great!
Filepond itself is able to process chunk upload - PovilasKorop recently made some video on how to process chunks in laravel but not in filament. [https://www.youtube.com/watch?v=xxo2uX5HoM8](url)

**Comment by aliaswpeu** [comment]
+1

**Comment by JavaDle** [comment]
+1

**Comment by Mahmoud217TR** [comment]
+1

**Comment by balex25** [comment]
+1

**Comment by matildasmets** [comment]
+1

**Comment by iRaziul** [comment]
Is there any chance of having this feature? 😅

**Comment by balex25** [comment]
[https://filamentphp.com/plugins/spykapp-uppy-upload-plugin](https://filamentphp.com/plugins/spykapp-uppy-upload-plugin) - what filament needed to have long time ago, **support file chunk upload + free**, give it a try 💪🏼

**Comment by Youssef-Amjad** [comment]
+1
